src/main/scala/com/fulcrumgenomics/umi/SimpleConsensusCaller.scala

```diff
@@ -0,0 +1,83 @@
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.umi
+
+import com.fulcrumgenomics.FgBioDef.forloop
+import com.fulcrumgenomics.util.NumericTypes.{LogProbability, PhredScore}
+
+/**
+  * A class that can be mixed in to call sequences represented as strings of the same length.
+  *
+  * If a non-ACGTN bases is encountered at any position in the strings, that position is not consensus called, and the
+  * character from the first sequenced is used.  All strings should have the same non-ACGTN character at this position.
+  *
+  * @param errorRatePreLabeling The error probability prior to adding the UMI.  By default, set to almost surely zero.
+  * @param errorRatePostLabeling The error probability post to adding the UMI.  By default, set to almost surely zero
+  * @param qError The adjusted error probability.  By default set to Q20.
+  */
+private[umi] class SimpleConsensusCaller(val errorRatePreLabeling: Byte = 90.toByte,
+                                         val errorRatePostLabeling: Byte = 90.toByte,
+                                         qError: PhredScore = 20.toByte
+                                        ) {
+  val pError: LogProbability = LogProbability.fromPhredScore(this.qError)
+  val pTruth: LogProbability = LogProbability.not(this.pError)
+
+  /** A consensus caller used to generate consensus UMI sequences */
+  private val consensusBuilder = new ConsensusCaller(
+    errorRatePreLabeling  = this.errorRatePreLabeling,
+    errorRatePostLabeling = this.errorRatePostLabeling
+  ).builder()
+
+  private val DnaBases = Set('A', 'C', 'G', 'T', 'N', 'a', 'c', 'g', 't', 'n')
+
+  /** Calls a simple consensus sequences from a set of sequences all the same length. */
+  def callConsensus(sequences: Seq[String]): String = {
+    require(sequences.nonEmpty, "Can't call consensus on an empty set of UMIs!")
+    require(sequences.forall(_.length == sequences.head.length), "Sequences must all have the same length")
+    val buffer = new StringBuilder
+
+    forloop (from=0, until=sequences.head.length) { i =>
+      this.consensusBuilder.reset()
+      var nonDna = 0
+      sequences.foreach { sequence =>
+        val char = sequence.charAt(i)
+        if (!this.DnaBases.contains(char)) {
+          nonDna += 1
+          // verify that all non-DNA bases are the same character
+          if (sequences.head.charAt(i) != char) {
```

Comment:
You could use `require(sequences.head.charAt(i) == char, ...)` here which is a little more compact.